src/realm/util/file.cpp

```diff
@@ -1152,6 +1152,11 @@ bool File::rw_lock(bool exclusive, bool non_blocking)
     // or the OS will deadlock when un-suspending the app.
     int mode = exclusive ? O_WRONLY | O_NONBLOCK : O_RDWR | O_NONBLOCK;
 
+    auto report_err = [path = m_fifo_path](int err, const char* msg) {
+        auto message = util::format("%1: %2, path = '%3'", msg, std::system_category().message(err), path);
+        throw RuntimeError(ErrorCodes::FileOperationFailed, message); // LCOV_EXCL_LINE
```

Comment:
i think we probably shouldn't worry about the cost of copying the string here, but should `message` be moved?

Reply:
I think not. RuntimeError takes a std::string_view as parameter.